x-pack/plugins/profiling/public/components/flame_graphs_view/get_impact_rows.ts

```diff
@@ -23,21 +24,26 @@ const CO2_PER_KWH = 0.92;
 const CORE_COST_PER_HOUR = 0.0425;
 
 export function getImpactRows({
+  countInclusive,
+  countExclusive,
   samples,
-  childSamples,
-  sampledTraces,
-  totalTraces,
+  totalSamples,
   totalSeconds,
 }: {
+  countInclusive: number;
+  countExclusive: number;
   samples: number;
-  childSamples: number;
-  sampledTraces: number;
-  totalTraces: number;
+  totalSamples: number;
   totalSeconds: number;
 }) {
-  const percentage = samples / sampledTraces;
-  const percentageNoChildren = (samples - childSamples) / sampledTraces;
-  const totalCoreSeconds = totalTraces / 20;
+  const samplingRate = samples / totalSamples;
+
+  const upscaledCountInclusive = countInclusive / samplingRate;
+  const upscaledCountExclusive = countExclusive / samplingRate;
+
+  const percentage = upscaledCountInclusive / totalSamples;
```

Comment:
This value differs from the inclusive CPU percentage displayed in the flamegraph tooltip.

I noticed the difference but I didn't investigate long enough to find the reason for the difference.

Reply:
I think this happens because the number of samples for the root frame is lower than the number of sampled traces (from the response). To calculate the sampling rate I'm using the latter but the flamegraph tooltip is using the former. Which one should be used to determine the percentage?

Reply:
Whatever `sampledTraces` is, please use `TotalTraces`, which should be the upscaled total number of samples.

Reply:
sorry I was not accurate there. I'm using `TotalTraces`, which is lower than the value of the root frame. Should the tooltip also use `TotalTraces` rather than the value of the root frame?

Reply:
Not sure what values these variables contain.
The returned JSON contains (example - from the MVP, last 30m):
```
  "TotalSeconds": 1800,
  "TotalTraces": 10205875,
  "SampledTraces": 81647
```
It is located at the end of the file. Here, `TotalTraces` is the correct value.
Can you give an example of the values you see ?

Reply:
The value for the root frame is in `.Samples[0]`. It's lower than `.SampledTraces` (in my environment).